### PR TITLE
Bug in F.p_value

### PR DESCRIPTION
**Pull request by v0dro**
@agarie: as I suspected theres a bug in the Ruby version of `Distribution::F.p_value`, which is causing a malfunction in statsample.

I've written a failing test for it, please check it out let me know what is to done, I'm not too familiar with F distributions. 

Here is the test result:

```

Failures:

  1) Distribution::F Distribution::F::Ruby_ it should behave like F engine should return correct p_value
     Failure/Error: instance_eval(&block)
       expected 20.784610000000026 to be within 0.0001 of 8.805687149846525
     Shared Example Group: "F engine" called from ./spec/f_spec.rb:76
     # ./spec/f_spec.rb:52:in `block (6 levels) in <top (required)>'
     # ./spec/f_spec.rb:49:in `each'
     # ./spec/f_spec.rb:49:in `block (5 levels) in <top (required)>'
     # ./spec/f_spec.rb:48:in `each'
     # ./spec/f_spec.rb:48:in `block (4 levels) in <top (required)>'
     # ./spec/f_spec.rb:47:in `each'
     # ./spec/f_spec.rb:47:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:32:in `instance_eval'
     # ./spec/spec_helper.rb:32:in `block in it_only_with_gsl'
```

### CODE REVIEW

File: spec/f_spec.rb [hunk 1]
```diff
@@ -3,68 +3,68 @@
...
-      pending("No #{@engine}.p_value")
+
+    it_only_with_gsl "should return correct p_value", :focus => true do
+      if @engine.respond_to? :p_value
+        [0.1,0.5,1,2,10,20,30].each do |f|
+          [2,5,10].each do|n2|
+            [2,5,10].each do |n1|
+              area = @engine.cdf(f,n1,n2)
+              @engine.p_value(area,n1,n2).should be_within(1e-4).of(GSL::Cdf.fdist_Pinv(area,n1,n2))
+              @engine.p_value(0.975, 5, 4.189092917592713).should be_within(
```
Comment (v0dro) [inline spec/f_spec.rb:52]: @agarie the values that you see here (the arguments to the `p_value` function) are the exact values that [statsample was passing into the function](https://github.com/v0dro/statsample/blob/master/lib/statsample/reliability/icc.rb#L344), which ultimately made it fail.

When I substitute `Distribution::F.p_value` with `GSL::Cdf.fdist_Pinv` in statsample, the test passes. So this is definitely a problem with the ruby implementation of `Distribution::F.p_value`/

File: distribution.gemspec [hunk 2]
```diff
@@ -23,4 +23,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'bundler'
   s.add_development_dependency 'rake'
   s.add_development_dependency "rspec", '~> 3.2'
+  s.add_development_dependency 'rb-gsl'
```
Comment (agarie) [inline distribution.gemspec]: Can you remove this line as well? I'm trying to find the bug and need to merge this patch, but don't want to add rb-gsl as a dependency, as I said in another thread. :)

**Comment by v0dro** [comment]
ok done

**Comment by agarie** [comment]
Good, thanks!

**Pull request merged**